### PR TITLE
BackupService backwards compatibility

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:

==COMMIT_MSG==
==COMMIT_MSG==

**Implementation Description (bullets)**:

**Testing (What was existing testing like?  What have you done to improve it?)**:

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:

### CODE REVIEW

File: atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/ExternalBackupPersister.java [hunk 1]
```diff
@@ -148,12 +152,24 @@ private <T> Optional<T> loadFromFile(AtlasService atlasService, File file, Class
...
-                    "Successfully loaded file",
-                    SafeArg.of("fileName", file.getName()),
-                    SafeArg.of("atlasService", atlasService));
-            return Optional.of(state);
+            try {
+                T state = OBJECT_MAPPER.readValue(file, clazz);
+                log.info(
+                        "Successfully loaded file",
+                        SafeArg.of("fileName", file.getName()),
+                        SafeArg.of("namespace", atlasService));
```
Comment (Jolyon-S) [inline atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/ExternalBackupPersister.java:160]: non-blocking: you've reverted this name

**Review by Jolyon-S — COMMENTED** [review]
gtg

**Review by Jolyon-S — APPROVED** [review]

File: atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/ExternalBackupPersister.java [hunk 2]
```diff
@@ -43,9 +48,12 @@ public class ExternalBackupPersister implements BackupPersister {
     private static final String FAST_FORWARD_TIMESTAMP_FILE_NAME = "fast-forward.timestamp";
 
     private final Function<AtlasService, Path> pathFactory;
+    private final Supplier<ObjectMapper> legacyObjectMapper;
 
     public ExternalBackupPersister(Function<AtlasService, Path> pathFactory) {
         this.pathFactory = pathFactory;
+        this.legacyObjectMapper = Suppliers.memoize(
+                () -> OBJECT_MAPPER.copy().setPropertyNamingStrategy(PropertyNamingStrategies.KEBAB_CASE));
```
Comment (Jolyon-S) [inline atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/ExternalBackupPersister.java:56]: is there any reason we can't just have a second static?

**Review by Jolyon-S — COMMENTED** [review]
one small question, otherwise looks good

File: atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/ExternalBackupPersister.java (hunk 2 above)
Comment (gsheasby) [reply atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/ExternalBackupPersister.java:56]: I guess we could - we most likely won't need to initialize this, but doing so can't be _that_ expensive, and it'd make the code a little simpler.

**Pull request merged**